baseapp/baseapp.go

```diff
@@ -526,53 +510,74 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (result sdk
 		ctx = ctx.WithMultiStore(msCache)
 	}
 
-	finalResult := sdk.Result{}
-	var logs []string
-	for i, msg := range msgs {
-		// Match route.
-		msgType := msg.Type()
-		handler := app.router.Route(msgType)
-		if handler == nil {
-			return sdk.ErrUnknownRequest("Unrecognized Msg type: " + msgType).Result()
-		}
+	// No need to run Msg checks when running CheckTx(), since CheckTx() only needs to validate that fees can be paid
+	// Then when DeliverTx() is called, this will run, preventing us from doing the same checks twice
+	if mode != runTxModeCheck {
 
-		result = handler(ctx, msg)
-
-		// Set gas utilized
-		finalResult.GasUsed += ctx.GasMeter().GasConsumed()
-		finalResult.GasWanted += result.GasWanted
+		// Get the Msg.
```

Comment:
maybe we can move this all into a new function since its specific to executing messages - will make this func easier to read